### ISSUE TITLE
Define additional disks with defined sizes to VMs.

### ISSUE THREAD

**Issue by erzetpe** [epiphany-platform/epiphany#139]
This will be done after creating new data model for Epiphany.

**Is your feature request related to a problem? Please describe.**
no

**Describe the solution you'd like**
Attach additional data_disks with specified mount points for each group of virtual machines.

**Describe alternatives you've considered**
None.

**Additional context**
None.

**Comment by mkyc** [comment]
There is [task](https://github.com/epiphany-platform/m-azure-basic-infrastructure/issues/13) in another repo for this in azure. We should probably have similar in aws module and consider it to be closed?

**Comment by ar3ndt** [comment]
Similar task in aws module created:
https://github.com/epiphany-platform/m-aws-basic-infrastructure/issues/27

**Comment by mkyc** [comment]
@erzetpe do you agree that we close this one?

**Comment by mkyc** [comment]
Actually this task is about disk sizes, and ones mentioned in 2 previous comments are about vm sizes. My mistake. In fact more related issue is [this one](https://github.com/epiphany-platform/m-azure-basic-infrastructure/issues/28).

**Comment by mkyc** [comment]
We should think about upgrade process before we start this task. It looks like we need spike to test upgrade process and probably later number of followup tasks.

**Comment by atsikham** [comment]
@toszo @to-bar @erzetpe @plirglo @rafzei I think we need a common discussion and I mention you as involved in support and as you are might be familiar with details.

How I would proceed with this issue (still being a bit out of context):
_Split this task into separate ones:_
1. Add support for disks/volumes customization in modules part.
2. Introduce new parameters in epicli (don't know in module or in standard one) to support installation/applying configuration with these disks, in the same place doing filesystem part configuration: mount, resizing, data movement. Probably this one should be splitted for different supported services where this functionality should be applied: kafka, rabbitmq, etc.
3. Make upgrade process the same as previously, but if it's necessary to add new disks, perform `epicli apply` first.

**Comment by toszo** [comment]
1. Ok
2. 3. Lets focus on backup-restore so we can easily apply such changes and restore data.

**Comment by atsikham** [comment]
Created tasks for the first point, see links above. About the 2nd and the 3rd points, I think that there should be an epic created for backup/restore of each component (who will identify/provide the list?) and we should make this task dependent on the whole epic and  tasks https://github.com/epiphany-platform/m-azure-basic-infrastructure/issues/42 and https://github.com/epiphany-platform/m-aws-basic-infrastructure/issues/37. After that `grooming` label can be removed here, but imo we should keep it open to test in the end of implementation.

@mkyc could you help with that?

**Comment by mkyc** [comment]
We can convert this task to epic, but I'd need some more detailed plan for this epic. AFAIU going with backup-restore way we would need to:
 - add support for separate disks in infrastructure creating VM's (my opinion is that this should be default and I do not need to configure anything to have separate disk used for "data")
 - ensure (for each component we have in classic epiphany) backup process works correctly
 - ensure (for each component as well) that we have either module which is able to restore from backup, or that classic epiphany is able to restore to VM's created with separate disks. 

Am I correct?

**Comment by atsikham** [comment]
> add support for separate disks in infrastructure creating VM's (my opinion is that this should be default and I do not need to configure anything to have separate disk used for "data")

We discussed that there should be possibility to configure, as each component requires different space amount and probably different mount points.

> ensure (for each component we have in classic epiphany) backup process works correctly

IMO that's correct.

> ensure (for each component as well) that we have either module which is able to restore from backup, or that classic epiphany is able to restore to VM's created with separate disks.

I think that it's enough to make it possible via classic Epiphany.

@sk4zuzu @to-bar @toszo Could you write your thoughts on that? I remember that last time you participated in discussion.

**Comment by sk4zuzu** [comment]
For me it's clear, we make the backup/restore process work flawlessly for every component and it has to be battle tested with big volumes of data. We have to know how much time it may take and what to do if restore process fails for some reason after let's say 3 hours and there are 5 hours left. During `epicli upgrade` we create and mount data disks, restore all backups and reconfigure all applications to use the volumes. This deserves a separate epic :+1: :innocent: , it's a lot of work and testing \\:heart_eyes:/.

**Comment by mkyc** [comment]
Just reminded myself that we have #1545 epic for backup/restore topics. 

Maybe we use this current task (139) only to implement option to add disks to VMs in classic epiphany?

**Comment by sk4zuzu** [comment]
@mkyc agreed. We make this task about adding optional disks to classic Epiphany clusters without moving any data. This mechanism should:
1. be generic enough to allow defining extra disks per component
2.

**Comment by seriva** [comment]
This issues should be changed to make sure it includes the best support for the cloud native storage, also as we AWS changes we might want to split this up in AWS and Azure separately.

**Comment by erzetpe** [comment]
Closing this as new tasks were created that describe this what needs to be done in current Epiphany more accurate.
